### DISCUSSION TITLE
Why is changesets erroring on a private and ignored package in devDependencies?

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [changesets/changesets#1158] [Q&A]
https://github.com/NullVoxPopuli/limber/actions/runs/4964806549/jobs/8885289122#step:5:34

Log:
```
 > changeset publish

🦋  error ValidationError: Some errors occurred when validating the changesets config:
🦋  error The package "ember-repl" depends on the ignored package "@nullvoxpopuli/limber-untyped", but "ember-repl" is not being ignored. Please add "ember-repl" to the `ignore` option.
🦋  error The package "limber-ui" depends on the ignored package "@nullvoxpopuli/limber-untyped", but "limber-ui" is not being ignored. Please add "limber-ui" to the `ignore` option.
🦋  error     at parse (/home/runner/work/limber/limber/node_modules/.pnpm/@changesets+config@2.3.0/node_modules/@changesets/config/dist/config.cjs.dev.js:277:11)
🦋  error     at Object.read (/home/runner/work/limber/limber/node_modules/.pnpm/@changesets+config@2.3.0/node_modules/@changesets/config/dist/config.cjs.dev.js:107:10)
🦋  error     at async run$2 (/home/runner/work/limber/limber/node_modules/.pnpm/@changesets+cli@2.26.1/node_modules/@changesets/cli/dist/cli.cjs.dev.js:1311:16) {
🦋  error   _error: Error
🦋  error       at new ExtendableError (/home/runner/work/limber/limber/node_modules/.pnpm/extendable-error@0.1.7/node_modules/extendable-error/bld/index.js:23:24)
🦋  error       at new ValidationError (/home/runner/work/limber/limber/node_modules/.pnpm/@changesets+errors@0.1.4/node_modules/@changesets/errors/dist/errors.cjs.dev.js:16:1)
🦋  error       at parse (/home/runner/work/limber/limber/node_modules/.pnpm/@changesets+config@2.3.0/node_modules/@changesets/config/dist/config.cjs.dev.js:277:11)
```

`ember-repl`, depends on `@nullvoxpopuli/limber-untyped` for TS stuff, it's dev only and in `ember-repl`- `devDependencies`.  

Same for `limber-ui`

`@nullvoxpopuli/limber-untyped` should not be published
 - it's private
 - it's ignored
 - it's in devDependencies
 
 am I missing something obvious?

**Comment by Andarist** [comment]
It seems that `getDependentsGraph` always checks against all dependency types so dev deps are not ignored by it. We'd appreciate a PR that would relax this check for ignored packages and one that would make `getDependentsGraph` and `getDependencyGraph` more flexible to ignore devDeps (or rather, one that would allow to customize the dependency types that the caller is interested in)

**Reply by m1aw** [reply]
What is the purpose of ignoring of having this warning anyway?
If pckg-a is bumped why would pckg-b need to be bumped too?

**Comment by adybionka** [comment]
Is there any update on this issue? It's currently blocking me from using this tool in our monorepo.

**Comment by sannajammeh** [comment]
We are currently blocked on this as well.
